pkg/apis/triggers/v1alpha1/trigger_binding_validation.go

```diff
@@ -30,11 +30,29 @@ func (t *TriggerBinding) Validate(ctx context.Context) *apis.FieldError {
 
 // Validate TriggerBindingSpec.
 func (s *TriggerBindingSpec) Validate(ctx context.Context) *apis.FieldError {
-	return validateParameters(s.Params)
+	if err := validateInputParams(s.InputParams); err != nil {
+		return err
+	}
+	if err := validateOutputParams(s.OutputParams); err != nil {
+		return err
+	}
+	return nil
+}
+
+func validateInputParams(params []v1alpha1.ParamSpec) *apis.FieldError {
+	// Ensure there aren't multiple params with the same name.
+	seen := map[string]struct{}{}
+	for _, p := range params {
+		if _, ok := seen[strings.ToLower(p.Name)]; ok {
+			return apis.ErrMultipleOneOf("spec.params")
```

Comment:
Should this be different between input and output params?

Reply:
I don't think it should be different because right now we're only checking that there aren't duplicates.